consensus/polybft/blockchain_wrapper.go

```
@@ -0,0 +1,191 @@
+func (p *blockchainWrapper) NewBlockBuilder(
+	parent *types.Header, coinbase types.Address) (blockBuilder, error) {
+	return NewBlockBuilder(&BlockBuilderParams{
+		Parent:      parent,
+		Coinbase:    coinbase,
+		ChainConfig: p.blockchain.Config(),
+		//ChainContext:  p.blockchain,
+		//TxPoolFactory: blockbuilder.NewEthTxPool(p.eth.TxPool()),
+		Executor: p.executor,
+		GasLimit: 10000000, // TODO: Nemanja - see what to do with this (p.eth.GenesisGasLimit(),)
+	}), nil
+}
+
+// GetSystemState is an implementation of blockchainBackend interface
+func (p *blockchainWrapper) GetSystemState(config *PolyBFTConfig, provider contract.Provider) SystemState {
+	return NewSystemState(config, provider)
+}
+
+func (p *blockchainWrapper) SubscribeEvents() blockchain.Subscription {
+	return p.blockchain.SubscribeEvents()
+}
+
+var _ contract.Provider = &stateProvider{}
+
+type stateProvider struct {
+	transition *state.Transition
+}
+
+// NewStateProvider initializes EVM against given state and chain config and returns stateProvider instance
+// which is an abstraction for smart contract calls
+func NewStateProvider(transition *state.Transition) contract.Provider {
+	return &stateProvider{transition: transition}
+}
+
+// Call implements the contract.Provider interface to make contract calls directly to the state
+func (s *stateProvider) Call(addr ethgo.Address, input []byte, opts *contract.CallOpts) ([]byte, error) {
+	result := s.transition.Call2(types.ZeroAddress, types.Address(addr), input, big.NewInt(0), 10000000)
+	if result.Failed() {
+		return nil, result.Err
+	}
+
+	return result.ReturnValue, nil
+}
+
+// Txn is part of the contract.Provider interface to make Ethereum transactions. We disable this function
+// since the system state does not make any transaction
+func (s *stateProvider) Txn(ethgo.Address, ethgo.Key, []byte) (contract.Txn, error) {
+	panic("we do not make transaction in system state")
```

Please don't introduce panics into the codebase. Why not return the error?

---

because it should be illegal to call this func but it is mandatory by the interface